test/extensions/clusters/common/logical_host_integration_test.cc

```diff
@@ -20,27 +20,29 @@ class LogicalHostIntegrationTest : public testing::Test, public HttpIntegrationT
 
   NiceMock<Network::MockDnsResolverFactory> dns_resolver_factory_;
   Registry::InjectFactory<Network::DnsResolverFactory> registered_dns_factory_;
+  std::shared_ptr<uint32_t> address_ptr_ = nullptr;
 };
 
 // Reproduces a race from https://github.com/envoyproxy/envoy/issues/32850.
 // The test is by mocking the DNS resolver to return multiple different
 // addresses, also config dns_refresh_rate to be extremely fast.
 TEST_F(LogicalHostIntegrationTest, LogicalDNSRaceCrashTest) {
+  address_ptr_ = std::make_shared<uint32_t>(0);
```

Comment:
you can use an `auto` here rather than a member var

Reply:
done